protocol/nats_jetstream/v3/options.go

```diff
@@ -0,0 +1,92 @@
+/*
+ Copyright 2021 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package nats_jetstream
+
+import (
+	"errors"
+
+	"github.com/nats-io/nats.go"
+	"github.com/nats-io/nats.go/jetstream"
+)
+
+var (
+	ErrNoFilterSubjects          = errors.New("no filter subjects were given")
+	ErrMoreThanOneStream         = errors.New("more than one stream for given filter subjects")
+	ErrNoConsumerConfig          = errors.New("no consumer config was given")
+	ErrMoreThanOneConsumerConfig = errors.New("more than one consumer config given")
+)
+
+// NatsOptions is a helper function to group a variadic nats.ProtocolOption into
+// []nats.Option that can be used by either Sender, Consumer or Protocol
+func NatsOptions(opts ...nats.Option) []nats.Option {
```

Comment:
nit: IMHO this helper can either by internal if we need it or removed because it increases the API for no real benefit